### ISSUE TITLE
Univariate Feature selection does not work on string features.

### ISSUE THREAD

**Issue by Neopazh** [krishnaik06/Feature-Selection-techniques#1]
Your tutorial on feature selection is the most convincing. I am a beginner, and I tried your univariate feature selection on a dataset from Kaggle (Titanic: Machine learning from disaster). I got stuck at the point where SelectbestK() gives me error that string cannot be converted to float. Please guide me through this error and also if you could make a detailed video or tutorial explaining some of the manipualtion needed to be made in raw data to make it feasible for feature selection.

**Comment by harshit-saraswat** [comment]
Hey @Neopazh, you must be knowing that most statistical and machine learning model can't work with String based features. So inorder to overcome that, we need to first encode the string based features into numerical features. You could use[ Label Encoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html) for that if the feature contains bi-categorical values. If the feature is having multiple categories then I'd advice you to first use [Label Encoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html) and then you can apply [One Hot Encoding](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html) or you could just use [pd.get_dummies() method](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.get_dummies.html). 
Once you've done this, you can easily use the [SelectKBest](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.SelectKBest.html).
I hope that helps. Thanks.
